File: docker/Dockerfile.go
FROM golang:1.22-alpine AS builder
WORKDIR /build
COPY src/go/go.mod src/go/go.sum ./
RUN go mod download
COPY src/go/ ./
RUN CGO_ENABLED=0 go build -o /hermesdeck ./cmd/hermesdeck/

FROM alpine:3.19
RUN apk add --no-cache ca-certificates
COPY --from=builder /hermesdeck /app/hermesdeck
EXPOSE 8080 50051
ENTRYPOINT ["/app/hermesdeck"]

File: docker/Dockerfile.py
FROM python:3.11-slim
RUN apt-get update && apt-get install -y --no-install-recommends git curl && rm -rf /var/lib/apt/lists/*
WORKDIR /app
COPY src/python/requirements.txt /app/
RUN pip install --no-cache-dir -r /app/requirements.txt
COPY src/python/hermesdeck_sidecar/ /app/hermesdeck_sidecar/
RUN mkdir -p /data
VOLUME ["/data"]
EXPOSE 50052
ENTRYPOINT ["python", "-m", "hermesdeck_sidecar"]
